docker/puppetserver/docker-entrypoint.d/80-ca.sh

```diff
@@ -43,13 +43,15 @@ else
   hocon -f /etc/puppetlabs/puppetserver/conf.d/ca.conf \
     set certificate-authority.allow-subject-alt-names "${CA_ALLOW_SUBJECT_ALT_NAMES}"
 
-  # Generate the same SSL directory that the PE installer creates.
-  #
-  # The steps in this file correspond to the steps in the PE installer.
-  #
-  # See https://github.com/puppetlabs/puppet-enterprise-modules/blob/kearney/modules/pe_install/manifests/prepare/certificates.pp
+  [ -d "$CADIR" ] && [ -f "$CADIR/ca_crt.pem" ]
+  new_ca_exists=$?
+
+  [ -d "$SSLDIR" ] && [ -f "$SSLDIR/ca_crt.pem" ]
+  old_ca_exists=$?
```

Comment:
These variable names don't match what you have below, `new_ca_exists` vs `new_ca_dir`

Reply:
:lol:

Reply:
Okay, I had some pretty basic mistakes where I started the variables updates in this script and then iterated in my shell with something similar and then came back to this script but things didn't line up. I think I've addressed them. Though it does beg the question of how best to test docker images.

Reply:
To test just do `make lint build test`

Reply:
Is this expected:
```
latest: Pulling from hadolint/hadolint
Digest: sha256:d6ee991347e4e0bd38e12b6bf915ac01c757e92e7da52c7a51a39e723ee7f68d
Status: Downloaded newer image for ghcr.io/hadolint/hadolint:latest
ghcr.io/hadolint/hadolint:latest
hadolint: Dockerfile: openBinaryFile: inappropriate type (is a directory)
make: *** [Makefile:51: lint] Error 1
```

Reply:
Weird:
```
justin@sellout:~/backups/Backup/code/work/server/puppetserver/docker$ sudo docker run --rm -v $(pwd)/puppetserver/Dockerfile:/Dockerfile -i ghcr.io/hadolint/hadolint:latest hadolint Dockerfile
justin@sellout:~/backups/Backup/code/work/server/puppetserver/docker$ echo $?
0
justin@sellout:~/backups/Backup/code/work/server/puppetserver/docker$ sudo make lint
latest: Pulling from hadolint/hadolint
Digest: sha256:d6ee991347e4e0bd38e12b6bf915ac01c757e92e7da52c7a51a39e723ee7f68d
Status: Image is up to date for ghcr.io/hadolint/hadolint:latest
ghcr.io/hadolint/hadolint:latest
hadolint: Dockerfile: openBinaryFile: inappropriate type (is a directory)
make: *** [Makefile:51: lint] Error 1
```

It looks like it gets past lint in Travis though and then builds the container, spams with a bunch of docker help output and then fails the test portion of that command.